drivers/SmartThings/zigbee-water-leak-sensor/src/sinope/init.lua

```diff
@@ -0,0 +1,70 @@
+-- Copyright 2022 SmartThings
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+local capabilities = require "st.capabilities"
+local zcl_clusters = require "st.zigbee.zcl.clusters"
+local IASZone = zcl_clusters.IASZone
+
+local SINOPE_TECHNOLOGIES_MFR_STRING = "Sinope Technologies"
+
+local generate_event_from_zone_status = function(driver, device, zone_status, zb_rx)
+  local wet_probe = 0x0031
+  local wet_probe_disconnected = 0x0033
+  local event
+
+  if zone_status.value == wet_probe or zone_status.value == wet_probe_disconnected then
```

Comment:
Is there a reason you _only_ emit wet if supervision and restore reports are enabled? Just from a quick look, the defaults, which check for whether alarm 1 or alarm 2 are set would also both trigger for ias zone status values of 0x0031 and 0x0033.

It doesn't seem like you need a sub-driver at all.

Reply:
You'll want to check this out: https://developer.smartthings.com/docs/devices/hub-connected/set-up-dev-env

Reply:
So the default driver checks for bits 0 and 1 to be set in the IAS Zone Status to emit a water leak event.

Your behavior here is to check bits 0, 1, 4, and 5. Bits 4 and 5 simply indicate the reporting behavior of the device, namely, whether the device sends periodic updates about this zone and whether it emits a separate event when the issue is cleared.

What was the bit value emitted by the device for which you were seeing a false positive? It doesn't seem like bits 4 and 5 should affect anything.

Reply:
I've made a mistake, I meant whenever the probe is disconnected. When the probe is disconnected, the device report this:
(Sinope Water Leak Sensor)> received Zigbee message: < ZigbeeMessageRx || type: 0x00, < AddressHeader || src_addr: 0x3397, src_endpoint: 0x01, dest_addr: 0x0000, dest_endpoint: 0x01, profile: 0x0104, cluster: IASZone >, lqi: 0xFF, rssi: -20, body_length: 0x0009, < ZCLMessageBody || < ZCLHeader || frame_ctrl: 0x19, seqno: 0x1D, ZCLCommandId: 0x00 >, < ZoneStatusChangeNotification || zone_status: 0x0032, extended_status: 0x00, zone_id: 0x01, delay: 0x0000 > > >

This kind of change is considered a wet accord to the main driver.

I did the change with a bitwise operator instead so only the first bit is considered.